lib/child_process.js

```diff
@@ -145,16 +145,12 @@ exports.execFile = function(file /*, args, options, callback*/) {
     windowsVerbatimArguments: !!options.windowsVerbatimArguments
   });
 
-  var encoding;
-  var _stdout;
-  var _stderr;
+  var encoding, stdoutEncoding, stderrEncoding;
```

Comment:
You don't need to store encodings, if a string is emitted you're already converting it to a Buffer right then and there so you know what the encoding for that string is at that time (it's set on the `StringDecoder` instance). When the stream finishes, you just check to see if a `StringDecoder` instance is set at that moment (`readableState.decoder`). If it is, then you just call `.toString(readableState.decoder.encoding)` on the concatenated Buffer. Otherwise just leave it as a Buffer.